sympy/core/sympify.py

```diff
@@ -206,12 +206,17 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     ``evaluate=False`` option will be added. Nested ``Add`` or ``Mul`` will
     be denested first. This is done via an AST transformation that replaces
     operators with their SymPy equivalents, so if an operand redefines any
-    of those operations, the redefined operators will not be used.
+    of those operations, the redefined operators will not be used. If
+    argument a is not a string, the mathematical expression is evaluated
+    before being passed to sympify, so adding evaluate=False will still
+    return the evaluated result of expression.
```

Comment:
I think we should add this as a subsection, like note or gotcha, something like that. 
It looks like users might encounter this if they don't pay attention. So, it might be better to add a subsection for it.

Reply:
> The same expression with evaluate argument is giving different results. One of the doctests failed because of that. I changed the output to make it work, but I dont understand why this happened

Because of this

Reply:
Here the inputs are same, both are not strings.

Reply:
That difference would happen when there are expression which are strings and some are not. Here both input on sympy online shell and example is not string. Still different results.

Reply:
Also should I add the difference of passing a string to sympify to the Pitfalls and Gotchas section of documentation. Here in the docstring I have mentioned it in notes section.